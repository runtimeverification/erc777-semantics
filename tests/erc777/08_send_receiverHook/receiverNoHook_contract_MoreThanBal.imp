test.preamble();

// send() to other address,  operatorSend() through default operator, operatorSend() from caller.
// contract holder destination with no hook (account 10), sending more than balance

test.makeCaller(3);
test.send(10, 200, "");
test.operatorSend(3, 10, 200, "", ""); //from caller

test.makeCaller(8);
test.operatorSend(3, 10, 200, "", ""); //through default operator
