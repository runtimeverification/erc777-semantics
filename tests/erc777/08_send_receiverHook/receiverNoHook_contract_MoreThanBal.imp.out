Changing the caller to 3
3 is sending 200 to 10 (data:)
REVERT
Balance of 3 is 30
Balance of 10 is 100
3 is sending 200 from 3 to 10 (data:""),  (operatorData:"")
REVERT
Balance of 3 is 30
Balance of 10 is 100
Changing the caller to 8
8 is sending 200 from 3 to 10 (data:""),  (operatorData:"")
REVERT
Balance of 3 is 30
Balance of 10 is 100
